lib/MusicBrainz/Server/Data/Search.pm

```diff
@@ -666,9 +666,7 @@ sub schema_fixup
         if (@languages) {
             $data->{languages} = [map {
                 MusicBrainz::Server::Entity::WorkLanguage->new({
-                    language => MusicBrainz::Server::Entity::Language->new({
-                        iso_code_3 => $_,
-                    }),
+                    language => $self->c->model('Language')->find_by_code($_),
```

Comment:
What do you think about removing this entire section and calling `$self->c->model('Language')->load_for_works(@entities);` [in external_search](https://github.com/yvanzo/musicbrainz-server/blob/521dab3/lib/MusicBrainz/Server/Data/Search.pm#L823-L828) so that we're only performing one query per search?

A comment explaining why we're doing this (the search doesn't include the name) seems appropriate, since we normally don't load information from the DB here.

Reply:
Although it performs more queries, most of these should be cached because of the small number of languages and their distribution for lyrics.

Removing this section doesn’t seem possible, as each lyrics `Language` still has to be linked to the related `Work` through `WorkLanguage`.

Calling `load_for_works` for indexed search opens to inconsistency between query and results display. It makes impossible to know which lyrics languages are currently indexed for a given work.

Reply:
I agree with your last point, but I don't believe `find_by_code` uses the cache currently. It calls `_get_by_key` which appears to always fetch from the database - I couldn't find any role that modifies its behavior to add caching either.

Reply:
Right, I still don't see how to _remove this entire section_, but I can make `_get_by_key` use cache. Would that be satisfactory? Or should we make search server indexing languages?

Reply:
Well, I believe `$self->c->model('Language')->load_for_works` does handle creating the `WorkLanguage` links, so I think you'd be able to remove the entire section in that case. But I agree with you that it creates inconsistencies. Changing `_get_by_key` to use the catch would be great!

Reply:
Exposing the language names in the WS might be a good idea since I'm not sure how people are supposed to get that information otherwise, if they don't have a DB. But it's a more involved change for sure, so just making the `find_by_code` queries cached sounds good to me.